packages/form-state-manager/src/utils/manager-api.ts

```diff
@@ -1,17 +1,24 @@
 import { FormEvent } from 'react';
 import set from 'lodash/set';
 
-import CreateManagerApi, { ManagerState, ManagerApi, AsyncWatcher, AsyncWatcherRecord } from '../types/manager-api';
+import CreateManagerApi, { ManagerState, ManagerApi, AsyncWatcher, AsyncWatcherRecord, Rerender } from '../types/manager-api';
 import AnyObject from '../types/any-object';
 import FieldConfig from '../types/field-config';
 import { formLevelValidator } from './validate';
 
+import merge from 'lodash/merge';
+import cloneDeep from 'lodash/cloneDeep';
+
 const isLast = (fieldListeners: AnyObject, name: string) => fieldListeners?.[name]?.count === 1;
 
 const addIfUnique = (array: Array<string>, item: string) => !array.includes(item) && array.push(item);
 
 const shouldExecute = (formLevel: boolean | undefined, fieldLevel: boolean | undefined) => (formLevel || fieldLevel) && fieldLevel !== false;
 
+type objectMapFunction = (value: any, key: any) => any;
+
+const traverseObject = (object: AnyObject, callback: objectMapFunction) => Object.keys(object).forEach((key) => callback(object[key], key));
```

Comment:
Any way to memoize this function? It could be the most expensive operation.

Reply:
I would keep optimization as the last step in our building. (We can then do some performance testing.)

Reply:
Can you add a TODO to the function? We will forget it. (I will)

Reply:
I did it. However, I have seen so many TODO comments I become immune to them. :smiley: 

Yeah, but we can focus on performance after we finalize the whole state construction. There are still so many blank spaces right now.